httpobs/scanner/analyzer/headers.py

```diff
@@ -4,6 +4,50 @@
 from httpobs.scanner.analyzer.utils import is_hpkp_preloaded, is_hsts_preloaded, only_if_worse
 
 
+def __parse_csp(csp_string: str) -> dict:
+    """
+    Decompose the CSP; could probably do this in one step, but it's complicated enough
+    Should look like:
+    {
+      'default-src': {'\'none\''},
+      'object-src': {'\'none\''},
+      'script-src': {'https://mozilla.org', '\'unsafe-inline\''},
+      'style-src': {'\'self\', 'https://mozilla.org'},
+      'upgrade-insecure-requests': {},
+    }
+    """
+
+    # Clean out all the junk
+    csp_string = csp_string.replace('\r', '').replace('\n', '').strip()
```

Comment:
This is probably fine, but personally I'd treat it like an array and use `csp_string.splitlines()`; it mirrors the nature of the data better.

Reply:
It actually shouldn't be split across lines. Carriage returns should be ignored, as it it's ideally all a single line. I've seen a lot of junk from various webservers, so this is my attempt to work aroundi t.